tests/unit/dataactvalidator/test_c23_award_financial_1.py

```diff
@@ -0,0 +1,51 @@
+from random import randint, choice
+from string import ascii_uppercase, ascii_lowercase, digits
+from tests.unit.dataactcore.factories.staging import AwardFinancialFactory, AwardProcurementFactory
+from tests.unit.dataactvalidator.utils import number_of_errors, query_columns
+
+_FILE = 'c23_award_financial_1'
+
+def test_column_headers(database):
+    expected_subset = {"row_number", "transaction_obligated_amou_sum", "federal_action_obligation_sum"}
+    actual = set(query_columns(_FILE, database))
+    assert expected_subset <= actual
+
+def test_success(database):
+    """ Test that a four digit object class with no flag is a success, and a three digit object class with a flag is a success"""
+    # Create a 12 character random piid
+    piid = ''.join(choice(ascii_uppercase + ascii_lowercase + digits) for i in range(12))
+    piid_two = ''.join(choice(ascii_uppercase + ascii_lowercase + digits) for i in range(12))
+    first_piid_row_one = AwardFinancialFactory(object_class = randint(1100,1999),
+        by_direct_reimbursable_fun = "d", transaction_obligated_amou = 1234, piid = piid)
```

Comment:
Are the `object_class` and `by_direct_reimbursable_fun` fields relevant to the test? If not, can we use the defaults? The less readers have to think about when reading the test, the easier they can understand it